kolibri/core/assets/src/views/sync/SelectAddressModalGroup/api.js

```diff
@@ -1,146 +1,60 @@
-    // locations that do not have the channel we are importing from.
-    if (withChannelId !== '') {
-      const locationsWithAvailabilityPromises = locations.map(location => {
-        // Need to wrap in normal promise, otherwise Promise.all will cause some of these
-        // to resolve as undefined
-        return new Promise(resolve => {
-          return channelIsAvailableAtLocation(withChannelId, location).then(isAvailable => {
-            resolve({ ...location, hasContent: isAvailable });
-          });
-        });
-      });
-      return Promise.all(locationsWithAvailabilityPromises);
-    }
-
-    // If channelId is not provided, then we are at top-level import workflow and do not
-    // disable any locations unless it is unavailable
-    return locations.map(location => ({ ...location, hasContent: location.available }));
+/**
+ * @param {string} facilityId
+ * @param {NetworkLocation} device
+ * @return {Promise<boolean>}
+ */
+export function facilityIsAvailableAtDevice(facilityId, device) {
+  return NetworkLocationResource.fetchFacilities(device.id).then(({ facilities }) => {
+    return Boolean(facilities.find(({ id }) => id === facilityId));
   });
 }
 
-function facilityIsAvailableAtLocation(facilityId, location) {
-  if (!location.available) {
-    return Promise.resolve(false);
-  }
-
-  return commonSyncElements.methods
-    .fetchNetworkLocationFacilities(location.id)
-    .then(({ facilities }) => {
-      return Boolean(facilities.find(({ id }) => id === facilityId));
-    });
-}
-
-function fetchAddressesWithFacility(facilityId = '', LocationResource = NetworkLocationResource) {
-  return LocationResource.fetchCollection({
+/**
+ * @param {string} channelId
+ * @param {NetworkLocation} device
+ * @return {Promise<boolean>}
+ */
+export function channelIsAvailableAtDevice(channelId, device) {
+  return RemoteChannelResource.fetchModel({
+    id: channelId,
+    getParams: {
+      baseurl: device.base_url,
+    },
     force: true,
-    getParams: { subset_of_users_device: false },
-  }).then(locations => {
-    if (facilityId !== '') {
-      const locationsWithAvailabilityPromises = locations.map(location => {
-        // Need to wrap in normal promise, otherwise Promise.all will cause some of these
-        // to resolve as undefined
-        return new Promise(resolve => {
-          return facilityIsAvailableAtLocation(facilityId, location).then(isAvailable => {
-            // NOTE: we're reusing 'hasContent' for both the facility/content cases for now
-            resolve({ ...location, hasContent: isAvailable });
-          });
-        });
-      });
-      return Promise.all(locationsWithAvailabilityPromises);
-    }
-
-    // If facilityId is not provided, then we are at the initial Facility Import workflow
-    // disable any locations unless it is unavailable/offline
-    return locations.map(location => ({ ...location, hasContent: location.available }));
```

Comment:
Consolidated this logic out of the API utilities, similar for both channels and facilities, into filtering composable structure